airbyte-ci/connectors/auto_merge/src/auto_merge/main.py

```diff
@@ -0,0 +1,178 @@
+
+if TYPE_CHECKING:
+    from github.Commit import Commit as GithubCommit
+    from github.File import File as GithubFile
+    from github.PullRequest import PullRequest
+    from github.Repository import Repository as GithubRepo
+
+logging.basicConfig()
+logging.getLogger().setLevel(logging.INFO)
+
+
+@contextmanager
+def github_client() -> Iterator[Github]:
+    client = None
+    try:
+        client = Github(auth=Auth.Token(GITHUB_TOKEN), seconds_between_requests=0)
+        yield client
+    finally:
+        if client:
+            client.close()
+
+
+def check_if_modifies_connectors_only(modified_files: Iterable[GithubFile]) -> bool:
+    """Check if all modified files are in CONNECTOR_PATH_PREFIXES
+
+    Args:
+        modified_files (Iterable[GithubFile]): All the modified files on a PR.
+
+    Returns:
+        bool: True if all modified files are in CONNECTOR_PATH_PREFIXES, False otherwise
+    """
+    for file in modified_files:
+        if not any(file.filename.startswith(prefix) for prefix in CONNECTOR_PATH_PREFIXES):
+            return False
+    return True
+
+
+def check_if_head_commit_passes_all_required_checks(head_commit: GithubCommit, required_checks: set[str]) -> bool:
+    """Required checks can be a mix of status contexts and check runs. A head commit is considered to pass all required checks if it has successful statuses and check runs for all required checks.
+
+    Args:
+        head_commit (GithubCommit): The head commit of the PR
+        required_checks (set[str]): The set of required passing checks
+
+    Returns:
+        bool: True if the head commit passes all required checks, False otherwise
+    """
+    successful_status_contexts = [commit_status.context for commit_status in head_commit.get_statuses() if commit_status.state == "success"]
+    successful_check_runs = [check_run.name for check_run in head_commit.get_check_runs() if check_run.conclusion == "success"]
+    successful_contexts = set(successful_status_contexts + successful_check_runs)
+    return required_checks.issubset(successful_contexts)
+
+
+def check_if_pr_is_auto_mergeable(head_commit: GithubCommit, pr: PullRequest, required_checks: set[str]) -> bool:
+    """A PR is considered auto-mergeable if:
+    - it has the AUTO_MERGE_LABEL
+    - it targets the BASE_BRANCH
+    - it touches only files in CONNECTOR_PATH_PREFIXES
+    - the head commit passes all required checks
+
+    Args:
+        head_commit (GithubCommit): The head commit of the PR
+        pr (PullRequest): The PR to check
+        required_checks (set[str]): The set of required passing checks
+
+    Returns:
+        bool: True if the PR is auto-mergeable, False otherwise
+    """
+    has_auto_merge_label = any(label.name == AUTO_MERGE_LABEL for label in pr.labels)
+    if not has_auto_merge_label:
+        logging.info(f"PR {pr.number} does not have the {AUTO_MERGE_LABEL} label")
+        return False
+    targets_main_branch = pr.base.ref == BASE_BRANCH
```

Comment:
Nit: you're declaring variables that are only used once on the very next line — let's just use the if statement directly?

I.e. `if not pr.base.ref == BASE_BRANCH:`. I don't think you will loose any readability with that.